xapian-core/weight/tfidfweight.cc

```diff
@@ -70,26 +74,48 @@ TfIdfWeight::name() const
 string
 TfIdfWeight::serialise() const
 {
-    return normalizations;
+    if (param_s > 0 || param_delta > 0) {
+	string result = serialise_double(param_s);
+	result += serialise_double(param_delta);
+	return result;
+    } else {
+	return normalizations;
+    }
 }
 
 TfIdfWeight *
-TfIdfWeight::unserialise(const string & s) const
+TfIdfWeight::unserialise(const string & str) const
 {
-    if (s.length() != 3)
-	throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
-    return new TfIdfWeight(s);
+    if (param_s > 0 || param_delta > 0) {
+	const char *ptr = str.data();
+	const char *end = ptr + str.size();
+	double s = unserialise_double(&ptr, end);
+	double delta = unserialise_double(&ptr, end);
+	if (rare(ptr != end))
+	    throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
+	return new TfIdfWeight(s, delta);
+    } else {
+	if (str.length() != 3)
+	    throw Xapian::SerialisationError("Extra data in TfIdfWeight::unserialise()");
+	return new TfIdfWeight(str);
+    }
 }
 
 double
-TfIdfWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount,
+TfIdfWeight::get_sumpart(Xapian::termcount wdf, Xapian::termcount doclen,
 			 Xapian::termcount) const
 {
     Xapian::doccount termfreq = 1;
+    double wqf_double = get_wqf();
```

Comment:
Should this not be just called `wqf`? Also, you probably don't want to pull it unless the normalisations are in play.